lib/BadgeOverlaySupport/BadgeOverlaySupport.js

```diff
@@ -0,0 +1,261 @@
+require('moonstone');
+
+var
+	kind = require('enyo/kind'),
+	Component = require('enyo/Component');
+var
+	Image = require('../Image'),
+	Icon = require('../Icon');
+
+/**
+* {@link moon.BadgeOverlaySupport} is a [mixin]{@glossary mixin} that may be applied
+* to any {@link moon.DataList} or {@link moon.DataGridList} item to provide an overlay
+* that is activated when the list contains items of type Video/Audio/Image.
+*
+* The BadgeOverlaySupport purpose is to show an icon badge on top of item to indicate
+* the type of item. Like for Video/Audio it will show a default play icon and for Image
+* there is nothing by default.
+*
+* The item has to define a
+* [badgeType]{@link moon.BadgeOverlaySupport#badgeType} MIME type of this
+* item. Possible values are 'video', 'audio' , 'image'. Defaulted as 'image'.
+*
+* ```javascript
+* {name: 'list', kind: 'moon.DataList', components: [
+*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video',
+*		kind: 'moon.ImageItem', bindings: [
+*			{from: '.model.title', to: '.label'},
+*			{from: '.model.description', to: '.text'},
+*			{from: '.model.coverSource', to: '.source'}
+*		]
+*	}
+* ]}
+* ```
+*
+* The item may define a
+* [badgeSrc]{@link moon.BadgeOverlaySupport#badgeSrc} URL to
+* override the default icon.
+*
+* ```javascript
+* {name: 'list', kind: 'moon.DataList', components: [
+*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video', badgeSrc: 'assets/my-icon.png',
+*		kind: 'moon.ImageItem', bindings: [
+*			{from: '.model.title', to: '.label'},
+*			{from: '.model.description', to: '.text'},
+*			{from: '.model.coverSource', to: '.source'}
+*		]
+*	}
+* ]}
+* ```
+*
+* The item may define a
+* [position]{@link moon.BadgeOverlaySupport#position} to
+* specify position of badge. Possible values are 'top' or 'bottom'. Defaulted as 'bottom'.
+*
+* ```javascript
+* {name: 'list', kind: 'moon.DataList', components: [
+*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video', position: 'top',
+*		kind: 'moon.ImageItem', bindings: [
+*			{from: '.model.title', to: '.label'},
+*			{from: '.model.description', to: '.text'},
+*			{from: '.model.coverSource', to: '.source'}
+*		]
+*	}
+* ]}
+* ```
+*
+* @mixin moon.BadgeOverlaySupport
+* @public
+*/
+BadgeOverlaySupport = exports.Support = {
```

Comment:
* `BadgeOverlaySupport` is created as a global var here. You don't need a local ref (afaik) so that can be removed.
* The convention in the framework has been to export the kind or mixin directly rather than as a member on the exported object. So, rather than `exports.Support`, just set the literal to exports

`module.exports = {`